packages/flame/lib/src/game/mixins/game.dart

```diff
@@ -110,16 +110,29 @@ mixin Game on Loadable implements Projector {
       ''');
     }
     _gameRenderBox = gameRenderBox;
+    onAttach();
   }
 
+  /// Called when the game has been attached this can be overriden
+  /// to add logic that requires that the game to be already attached
+  /// to the widget tree
+  void onAttach() {}
+
   /// Marks game as no longer attached to any Flutter widget tree.
   ///
   /// Should not be called manually.
   void detach() {
     _gameRenderBox = null;
     _size = null;
+
+    onDetach();
   }
 
+  /// Called after the game has left the widget tree
+  /// this can be overriden to add logic that requires the game
+  /// not be on the flutter widget tree anymore
```

Comment:
```suggestion
  /// Called after the game has left the widget tree.
  /// This can be overriden to add logic that requires the game
  /// not be on the flutter widget tree anymore.
```